kratos/includes/model_part.h

```diff
@@ -568,6 +599,111 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
+     */
+    void AddMasterSlaveConstraints(std::vector<IndexType> const& MasterSlaveConstraintIds);
+
+    /** Creates a new master-slave constraint in the current modelpart.
+     */
+    MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint(std::string ConstraintName, IndexType Id, DofsVectorType& rMasterDofsVector,
+                                                                                    DofsVectorType& rSlaveDofsVector,
+                                                                                    MatrixType RelationMatrix,
+                                                                                    VectorType ConstantVector);
+
+    MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint(std::string ConstraintName, IndexType Id, NodeType& rMasterNode,
+                                                                                    VariableType& rMasterVariable,
+                                                                                    NodeType& rSlaveNode,
+                                                                                    VariableType& rSlaveVariable,
+                                                                                    double Weight,
+                                                                                    double Constant);
+
+    MasterSlaveConstraintType::Pointer CreateNewMasterSlaveConstraint(std::string ConstraintName, IndexType Id, NodeType& rMasterNode,
+                                                                                    VariableComponentType& rMasterVariable,
+                                                                                    NodeType& rSlaveNode,
+                                                                                    VariableComponentType& rSlaveVariable,
+                                                                                    double Weight,
+                                                                                    double Constant);
+
+    /** Remove the master-slave constraint with given Id from mesh with ThisIndex in this modelpart and all its subs.
+    */
+    void RemoveMasterSlaveConstraint(IndexType MasterSlaveConstraintId);
+
+    /** Remove given master-slave constraint from mesh with ThisIndex in this modelpart and all its subs.
+    */
+    void RemoveMasterSlaveConstraint(MasterSlaveConstraintType& ThisMasterSlaveConstraint);
+
+    /** Remove given master-slave constraint from mesh with ThisIndex in this modelpart and all its subs.
+    */
+    void RemoveMasterSlaveConstraint(MasterSlaveConstraintType::Pointer pThisMasterSlaveConstraint);
+
+    /** Remove the master-slave constraint with given Id from mesh with ThisIndex in parents, itself and children.
+    */
+    void RemoveMasterSlaveConstraintFromAllLevels(IndexType MasterSlaveConstraintId);
```

Comment:
Can we avoid this function by doing `GetRootModelPart().RemoveMasterSlaveConstraint()`?

Reply:
Yes, but it is not what all the developers can deduce by themselves. FromAllLevels gives them better description than the  replacing code.